.github/workflows/build-project.yml

```diff
@@ -0,0 +1,84 @@
+name: build-project-all-platforms
+
+on:
+  workflow_dispatch:
+    # Inputs the workflow accepts.
+    inputs:
+      name:
+        # Friendly description to be shown in the UI instead of 'name'
+        description: 'Building and uploading paperless imagefor amd64/armv7/64'
+        # Default value if no value is explicitly provided
+        default: 'beta'
+        # Input has to be provided for the workflow to run
+        required: true
+jobs:
+  # build and push image to docker hub.
+  build-project:
+    #if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/ngx-project')
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Checkout software
+        uses: actions/checkout@v2
+      -
+        name: Prepare Docker
+        id: prepare
+        run: |
+          IMAGE_NAME=ghcr.io/schnuffle/paperless-docker
+          TAGS=${IMAGE_NAME}:latest
+          INSPECT_TAG=${IMAGE_NAME}:latest
+          echo ::set-output name=tags::${TAGS}
+          echo ::set-output name=inspect_tag::${INSPECT_TAG}
+      - 
```

Comment:
Add the metadata-action here. I am assuming we are using semver. 

```
      - name: Docker meta
        id: meta
        uses: docker/metadata-action@v3
        with:
          images: ghcr.io/${{ github.repository_owner }}/paperless-docker
          tags: |
            type=ref,event=branch
            type=ref,event=pr
            type=semver,pattern={{version}}
            type=semver,pattern={{major}}.{{minor}}
```

Reply:
This is a good note but I think it'd be best to put this in a new PR to minimize the diff here.

Reply:
I've actually included the `metadata-action`, but please check my attempts to generate the right values and such from tags and branches